### PR TITLE
SERVICES-627_Consolidate the logout functionality

### PR DESCRIPTION
**Pull request by sqreek**
Add AuthService and HeliosAuthService as implementation
Move logout call from Helios extension into User.php
remove mercury access token

ping @drsnyder @nmonterroso @harnash @ArturKlajnerok

### CODE REVIEW

**Comment by sqreek** [comment]
Tests was fallen becuse of jenkins (http://qa-s2.wikia-prod:8080/job/unitTests_js_branches/16696/console)

Now they passes but no sending check (http://qa-s2.wikia-prod:8080/job/unitTests_js_branches/16711/)

**Comment by harnash** [comment]
Seems fine to me.

File: lib/Wikia/src/Service/User/Auth/HeliosAuthService.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	}
+
+	/**
+	 * Call helios invalidate token.
+	 */
+	private function invalidateAccessTokenInHelios() {
+		$request = \RequestContext::getMain()->getRequest();
+		$this->heliosClient->invalidateToken( HeliosHelper::getAccessToken( $request ) );
+	}
+}
```
Comment (drsnyder) [inline lib/Wikia/src/Service/User/Auth/HeliosAuthService.php:38]: Isn't there also a `logout` required on the user object?

File: lib/Wikia/src/Service/User/Auth/AuthServiceBase.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+			return null;
+		}
+
+		$user->load();
+		if ( $user->getId() == 0 ) {
+			return null;
+		}
+
+		return $user->isBlocked();
+	}
```
Comment (drsnyder) [inline lib/Wikia/src/Service/User/Auth/AuthServiceBase.php:29]: I don't think this is necessary here since it's on `User`.

**Pull request closed without merging**